File: ECOO-18-1-1.c
#include <stdio.h.>
#include <stdlib.h>

int main() {
    int playDuration;
    int numDays;

    scanf("%d %d", &playDuration, &numDays);

    int queuedBoxes;
    int currentBox;
    int timeLeft;
    char dayResult;
    for(int i = 0; i < numDays; i++) {
        scanf(" %c", &dayResult);
        if(dayResult == 'B') {
            if(currentBox == 1) {
                queuedBoxes++;
            } else {
                currentBox = 1;
                timeLeft = playDuration;
            }
        }
        if(currentBox == 1) {
            timeLeft--;
        } else {
            if(queuedBoxes > 0) {
                queuedBoxes--;
                currentBox = 1;
                timeLeft = playDuration-1;
            }
        }
        if(timeLeft <= 0) {
            currentBox = 0;
        }
    }
    timeLeft += playDuration*queuedBoxes;
    printf("%d\n", timeLeft);

    return 0;
}

File: ECOO-18-1-1.py
file = open("C:\\Users\\Kevin\\Downloads\\ECOO2018\\Round 1\\DATA\\DATA11.txt")
for test in range(10):
    playLength, numDays = map(int, file.readline().strip().split())
    queuedBoxes, currentBox = 0, 0
    timeLeft = 0

    for i in range(numDays):
        if file.readline().strip() == "B":
            if currentBox == 0:
                currentBox = 1
                timeLeft = playLength-1
            else:
                timeLeft -= 1
                queuedBoxes += 1
        else:
            if currentBox == 1:
                timeLeft -= 1
        if timeLeft < 0:
            if queuedBoxes > 0:
                queuedBoxes -= 1
                timeLeft = playLength-1
            else:
                currentBox = 0
                timeLeft = 0
    timeLeft += queuedBoxes*playLength
    print timeLeft
